resources/codegen_inputs/trak.rs

```diff
@@ -0,0 +1,43 @@
+#![parse_module(read_fonts::tables::trak)]
+
+/// The [tracking (trak)](https://developer.apple.com/fonts/TrueType-Reference-Manual/RM06/Chap6trak.html) table.
+#[tag = "trak"]
+table Trak {
+    /// Version number of the tracking table (0x00010000 for the current version).
+    version: MajorMinor,
+    /// Format of the tracking table (set to 0).
+    format: u16,
+    /// Offset from start of tracking table to TrackData for horizontal text (or 0 if none).
+    #[nullable]
+    horiz_offset: Offset16<TrackData>,
+    /// Offset from start of tracking table to TrackData for vertical text (or 0 if none).
+    #[nullable]
+    vert_offset: Offset16<TrackData>,
+    /// Reserved. Set to 0.
+    #[skip_getter]
+    #[compile(0)]
+    reserved: u16,
+}
+
+/// The tracking data table.
+table TrackData {
+    /// Number of separate tracks included in this table.
+    n_tracks: u16,
+    /// Number of point sizes included in this table.
+    n_sizes: u16,
+    /// Offset from the start of the tracking table to the start of the size subtable.
+    size_table_offset: u32,
+    /// Array of TrackTableEntry records.
+    #[count($n_tracks)]
+    track_table: [TrackTableEntry],
+}
```

Comment:
Do we not support sizeTable[nSizes] as a subsequent array? Maybe we should? (could be deferred to an issue referring to this)

Reply:
We do support that but this table contains an explicit offset to the size table and we've been bitten by this before when assuming we can ignore it. In particular, Apple seems to like tables that are aligned to 4 bytes so it's conceivable that the size table might not directly follow the tracks.